Kernel/Bus/VirtIO/Device.cpp

```diff
@@ -125,6 +125,20 @@ UNMAP_AFTER_INIT void Device::initialize()
     }
 
     if (m_use_mmio) {
+        // Allocate backing kernel memory regions for BARs
+        for (auto bar = 0u; bar < sizeof(m_mmio)/sizeof(*m_mmio); ++bar) {
+            auto& mapping = m_mmio[bar];
+            mapping.size = PCI::get_BAR_space_size(pci_address(), bar);
+            if (!mapping.size)
+                continue;
+            auto region_or_error = MM.allocate_kernel_region(PhysicalAddress(page_base_of(PCI::get_BAR(pci_address(), bar))), Memory::page_round_up(mapping.size), "VirtIO MMIO", Memory::Region::Access::ReadWrite, Memory::Region::Cacheable::No);
+            if (region_or_error.is_error()) {
+                dbgln_if(VIRTIO_DEBUG, "{}: Failed to map bar {} - (size={}) {}", VirtIO::determine_device_class(pci_address()), bar, mapping.size, region_or_error.error());
+                return InitializationState::OutOfMemory;
```

Comment:
This method returns void, so this wont compile

Reply:
Oh, It merged into a newer version of the file. I'll fix in next revision, though we should have some way of signalling failure to initialize.

Reply:
Oh, for sure.